runtime/src/test_utils.rs

```diff
@@ -1289,6 +1295,17 @@ impl<BS: Blockstore> Runtime for MockRuntime<BS> {
     fn read_only(&self) -> bool {
         false
     }
+
+    fn payable(&mut self) -> TokenAmount {
+        self.require_in_call();
+
+        let expected_payment = self.expectations.borrow_mut().expect_value.take();
+        if let Some(payment) = expected_payment.clone() {
+            assert_eq!(payment, self.message().value_received(), "unexpected amount received");
```

Comment:
This is confusing. This is kind of testing that the test code is consistent, unnecessarily. The payment amount should be either in the expectation, or the message value received, not both. I think only the latter makes sense, so then the expectation of calling payable() is just a boolean.